examples/nimbus/src/lib.rs

```diff
@@ -0,0 +1,71 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use std::path::Path;
+#[derive(Debug, Clone)]
+pub struct ApplicationContext {
+    app_id: Option<String>,
+    app_version: Option<String>,
+    locale: Option<String>,
+    device_manufacturer: Option<String>,
+    device_model: Option<String>,
+    region: Option<String>,
+    debug_tag: Option<String>,
+}
+
+pub struct Experiment {
+    slug: String,
+    user_facing_name: String,
+    user_facing_description: String,
+}
+
+pub struct Config {
+    server_url: Option<String>,
+    uuid: Option<String>,
+    collection_name: Option<String>,
+    bucket_name: Option<String>,
+}
+
+#[derive(Debug, thiserror::Error)]
+pub enum ExperimentError {
+    #[error("Error from the network")]
+    NetworkError,
+    #[error("Error retrieving from the database")]
+    DbError,
+}
+
+type Result<T, E = ExperimentError> = std::result::Result<T, E>;
+pub struct Experiments {}
+
+impl Experiments {
+    pub fn new<P: AsRef<Path>>(_: ApplicationContext, _: P, _: Option<Config>) -> Result<Self> {
+        unimplemented!()
+    }
+
+    pub fn get_experiment_branch<S: Into<String>>(&self, _: S) -> Option<String> {
```

Comment:
This is surprising. Why not `String` or `&str`?

```rust
<S: Into<String>>
```

Reply:
Aaah this was muscle memory from when we had asymmetry, and I wanted my `todolist` to be able to take in either `&str` or `String`. Having as `String` is better though, thanks!!